FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/src/plaintext_freertos.c

```diff
@@ -72,6 +78,22 @@ BaseType_t Transport_FreeRTOS_Connect( NetworkContext_t * pNetworkContext,
         }
     }
 
+    if( socketStatus == 0 )
+    {
+        /* Set socket send timeout. */
+        transportTimeout = pdMS_TO_TICKS( sendTimeoutMs );
+        socketStatus = FreeRTOS_setsockopt( tcpSocket,
+                                            0,
+                                            FREERTOS_SO_SNDTIMEO,
+                                            &transportTimeout,
+                                            sizeof( TickType_t ) );
+
+        if( socketStatus != 0 )
+        {
+            LogError( ( "Failed to set socket send timeout." ) );
+        }
+    }
```

Comment:
Can setting the send and recv timeouts be moved to after a successful FreeRTOS_Connect() call? I was seeing an issue when trying to connect with a local broker if timeouts are set before FreeRTOS_Connect. It may be a TCP issue, following up with them, but we will have to change the order to unblock our demos.

Reply:
Moved to after the connect call. I also removed the error logs for them since the calls cannot fail.